pkg/actuators/machine/vm.go

```diff
@@ -14,6 +15,146 @@ import (
 	clientpkg "github.com/openshift/machine-api-provider-nutanix/pkg/client"
 )
 
+// validateVMConfig verifies if the machine's VM configuration is valid
+func validateVMConfig(mscp *machineScope) field.ErrorList {
+	errList := field.ErrorList{}
+	fldPath := field.NewPath("spec", "providerSpec", "value")
+
+	var err error
+	var errMsg string
+
+	// verify the cluster configuration
+	switch mscp.providerSpec.Cluster.Type {
+	case machinev1.NutanixIdentifierName:
+		if mscp.providerSpec.Cluster.Name == nil || *mscp.providerSpec.Cluster.Name == "" {
+			errList = append(errList, field.Required(fldPath.Child("cluster", "name"), "Missing cluster name"))
+		} else {
+			clusterName := *mscp.providerSpec.Cluster.Name
+			clusterRefUuidPtr, err := findClusterUuidByName(mscp.nutanixClient, clusterName)
+			if err != nil {
+				errMsg = fmt.Sprintf("Failed to find cluster with name %q. error: %v", clusterName, err)
+				errList = append(errList, field.Invalid(fldPath.Child("cluster", "name"), clusterName, errMsg))
+			} else {
+				mscp.providerSpec.Cluster.Type = machinev1.NutanixIdentifierUUID
+				mscp.providerSpec.Cluster.UUID = clusterRefUuidPtr
+			}
+		}
+	case machinev1.NutanixIdentifierUUID:
+		if mscp.providerSpec.Cluster.UUID == nil || *mscp.providerSpec.Cluster.UUID == "" {
+			errList = append(errList, field.Required(fldPath.Child("cluster", "uuid"), "Missing cluster uuid"))
+		} else {
+			clusterUUID := *mscp.providerSpec.Cluster.UUID
+			if _, err = mscp.nutanixClient.V3.GetCluster(clusterUUID); err != nil {
+				errMsg = fmt.Sprintf("Failed to find cluster with uuid %v. error: %v", clusterUUID, err)
+				errList = append(errList, field.Invalid(fldPath.Child("cluster", "uuid"), clusterUUID, errMsg))
+			}
+		}
+	default:
+		errMsg = fmt.Sprintf("Invalid cluster identifier type, the valid is %q or %q.", machinev1.NutanixIdentifierName, machinev1.NutanixIdentifierUUID)
```

Comment:
Substitute "the valid is" with "valid types are"

Reply:
Changed